packages/user_packages/iotcc_mqtt.py

```diff
@@ -79,7 +80,17 @@ def run(self, registry):
             self.iotcc.set_system_properties(self.iotcc_edge_system, registry.get("system_properties"))
             # Set the properties for edge system as key:value pair, you can also set the location
             # by passing the latitude and longitude as a property in the user package
-            self.iotcc.set_properties(self.iotcc_edge_system,{"key1": "value1", "key2": "value2"})
+            # If the set_properties or register call fails due to DCC_Comms Publish exception
+            # the optional retry mechanism can be implemented in the following way
+            attempts = 0
+            while attempts < 3:
```

Comment:
This doesn't seem optional, and isn't this going to be an infinite loop when the call succeeds since we don't break out of the while on success? If it fails 3 times the loop will break when the condition is met, though... and we don't log that failure?